docs/template_plugin/tests/functional/op_reference/convert.cpp

```diff
@@ -0,0 +1,432 @@
+// Copyright (C) 2018-2021 Intel Corporation
+// SPDX-License-Identifier: Apache-2.0
+//
+
+#include <gtest/gtest.h>
+
+#include <ie_core.hpp>
+#include <ie_ngraph_utils.hpp>
+#include <ngraph/ngraph.hpp>
+#include <shared_test_classes/base/layer_test_utils.hpp>
+#include <tuple>
+
+#include "base_reference_test.hpp"
+
+using namespace ngraph;
+using namespace InferenceEngine;
+
+struct ConvertParams {
+    template <class IT, class OT>
+    ConvertParams(const ngraph::PartialShape& shape, const ngraph::element::Type& iType, const ngraph::element::Type& oType, const std::vector<IT>& iValues,
+                  const std::vector<OT>& oValues, size_t iSize = 0, size_t oSize = 0)
+        : pshape(shape), inType(iType), outType(oType), inputData(CreateBlob(iType, iValues, iSize)), refData(CreateBlob(oType, oValues, oSize)) {}
+    ngraph::PartialShape pshape;
+    ngraph::element::Type inType;
+    ngraph::element::Type outType;
+    InferenceEngine::Blob::Ptr inputData;
+    InferenceEngine::Blob::Ptr refData;
+};
+
+class ReferenceConvertLayerTest : public testing::TestWithParam<ConvertParams>, public CommonReferenceTest {
+public:
+    void SetUp() override {
+        auto params = GetParam();
+        function = CreateFunction(params.pshape, params.inType, params.outType);
+        inputData = {params.inputData};
+        refOutData = {params.refData};
+    }
+
+private:
+    static std::shared_ptr<Function> CreateFunction(const PartialShape& input_shape, const element::Type& input_type,
+                                                    const element::Type& expected_output_type) {
+        const auto in = std::make_shared<op::Parameter>(input_type, input_shape);
+        const auto convert = std::make_shared<op::Convert>(in, expected_output_type);
+        return std::make_shared<Function>(NodeVector {convert}, ParameterVector {in});
+    }
+};
+
+TEST_P(ReferenceConvertLayerTest, CompareWithHardcodedRefs) {
+    Exec();
+}
+
+INSTANTIATE_TEST_SUITE_P(
```

Comment:
Generated test names are defaulting to `smoke_Convert_With_Hardcided_Refs/ReferenceConvertLayerTest.CompareWithHardcodedRefs/1` which tell nothing about parameters, hence if one test fails you don't know for which specific use case. I think names should include parameters value for ease of future debugging. 

Another thing is `Available devices for TEMPLATE:` in output. Looks like debug output, shouldn't we remove it?

```
jdanieck@ubuntu-ngraph:~/openvino/build$ ../bin/intel64/Release/templateFuncTests --gtest_filter=*ReferenceConvert*
Note: Google Test filter = *ReferenceConvert*
[==========] Running 175 tests from 1 test suite.
[----------] Global test environment set-up.
[----------] 175 tests from smoke_Convert_With_Hardcided_Refs/ReferenceConvertLayerTest
[ RUN      ] smoke_Convert_With_Hardcided_Refs/ReferenceConvertLayerTest.CompareWithHardcodedRefs/0
Available devices for TEMPLATE:
    
[       OK ] smoke_Convert_With_Hardcided_Refs/ReferenceConvertLayerTest.CompareWithHardcodedRefs/0 (43 ms)
[ RUN      ] smoke_Convert_With_Hardcided_Refs/ReferenceConvertLayerTest.CompareWithHardcodedRefs/1
Available devices for TEMPLATE:

...

[       OK ] smoke_Convert_With_Hardcided_Refs/ReferenceConvertLayerTest.CompareWithHardcodedRefs/173 (2 ms)
[ RUN      ] smoke_Convert_With_Hardcided_Refs/ReferenceConvertLayerTest.CompareWithHardcodedRefs/174
Available devices for TEMPLATE:
    
[       OK ] smoke_Convert_With_Hardcided_Refs/ReferenceConvertLayerTest.CompareWithHardcodedRefs/174 (1 ms)
[----------] 175 tests from smoke_Convert_With_Hardcided_Refs/ReferenceConvertLayerTest (289 ms total)

[----------] Global test environment tear-down
[==========] 175 tests from 1 test suite ran. (290 ms total)
[  PASSED  ] 175 tests.
```

Reply:
Fixed